llvm/lib/MC/MCRegisterInfo.cpp

```diff
@@ -20,6 +20,85 @@
 
 using namespace llvm;
 
+namespace {
+/// MCRegAliasIterator enumerates all registers aliasing Reg.  This iterator
+/// does not guarantee any ordering or that entries are unique.
+class MCRegAliasIteratorImpl {
```

Comment:
Stray thought: this class is used only in one place and could be inlined.

Reply:
I would prefer keeping it outlined, the iteration logic isn't trivial and it feels better when it's encapsulated like this.
Most of it will likely just get inlined anyway.

Reply:
It would be just three nested loops as I can see it. Anyway, this was just a thought.